### PR TITLE
Add bully package

### PR DESCRIPTION
**Pull request by p4u**
Bully is a new implementation of the WPS brute force attack, written in
C. It is conceptually identical to other programs, in that it exploits
the (now well known) design flaw in the WPS specification. It has
several advantages over the original reaver code.

Signed-off-by: p4u <p4u@dabax.net>

Maintainer: p4u
Compile tested: OpenWRT master
Run tested: tested and works

### CODE REVIEW

**Comment by aparcar** [comment]
is it possible to use codeload?

**Comment by aparcar** [comment]
also real names in signature are required afaik

File: net/bully/Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=bully
+PKG_VERSION:=1.1
+PKG_RELEASE:=31
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/aanarchyy/bully/archive/$(PKG_VERSION).tar.gz
```
Comment (diizzyy) [inline net/bully/Makefile]: URL shouldn't contain file name

File: net/bully/Makefile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=bully
+PKG_VERSION:=1.1
+PKG_RELEASE:=31
```
Comment (diizzyy) [inline net/bully/Makefile]: PKG_RELEASE start with 1, not 31

File: net/bully/Makefile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  DEPENDS:=+libpcap +pixiewps
+endef
+
+define Package/bully/description
+  Brute force attack against WPS, that actually works
+endef
+
+CONFIGURE_PATH:=src
+MAKE_PATH:=src
+TARGET_CFLAGS+=$(TARGET_CPPFLAGS)
```
Comment (diizzyy) [inline net/bully/Makefile]: Why is this needed?

File: net/bully/Makefile (hunk 1 above)
Comment (neheb) [reply net/bully/Makefile]: I prefer that the latest revision is used instead of version 1.1. It's 2016 vs. 2017.

File: net/bully/Makefile (hunk 2 above)
Comment (neheb) [reply net/bully/Makefile]: I think the confusion here is that he wants version 1.1 + 31 commits.

**Comment by neheb** [comment]
ping @p4u

**Comment by diizzyy** [comment]
Will close in 7d (2w timeout from last review comment)

File: net/bully/Makefile (hunk 2 above)
Comment (p4u) [reply net/bully/Makefile]: I took it from here https://github.com/aanarchyy/bully/blob/master/version.mk

Not sure either why 31, but I'll change to 1.

File: net/bully/Makefile (hunk 1 above)
Comment (p4u) [reply net/bully/Makefile]: Then `PKG_SOURCE` should be `$(PKG_VERSION).tar.gz`
And `PKG_SOURCE URL` should be `https://github.com/aanarchyy/bully/archive/`

If not, wget wont find the file in github.

File: net/bully/Makefile (hunk 3 above)
Comment (p4u) [reply net/bully/Makefile]: I think that's a garbage from the original Makefile https://github.com/aanarchyy/bully/blob/master/Makefile

File: net/bully/Makefile (hunk 1 above)
Comment (p4u) [reply net/bully/Makefile]: Ok, so then let's use "master"

**Comment by p4u** [comment]
Sorry for the delay. I'll make the suggested changes, test it and amend to PR.

File: net/bully/Makefile (hunk 1 above)
Comment (diizzyy) [reply net/bully/Makefile]: ```$(PKG_VERSION).tar.gz``` is a no go, either use codeload (grep the tree) or git over https.

Comment (neheb) [reply net/bully/Makefile]: PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://codeload.github.com/aanarchyy/bully/tar.gz/$(PKG_VERSION)?

**Comment by aparcar** [comment]
@p4u 

```
Author name (p4u) need to be your real name 'firstname lastname'
Commit subject line MUST start with '<package name>: ' (Add bully package)
```

**Comment by BKPepe** [comment]
Closing. If you will find a time to look at it, feel free to reopen PR.

**Pull request closed without merging**